### ISSUE TITLE
-Wmissing-designated-field-initializers ignores members of the base classes

### ISSUE THREAD

**Issue by jonathanpoelen** [llvm/llvm-project#210558]
When a designated initializer is used, Clang ignores the members of the base classes and does not issue a warning. Even though there is no way to initialize the members of the base classes in this case, I think there should be a warning.

```cpp
struct BBase
{
    int x; // non-default-inititialized
};

struct B : BBase
{
  int a = 0;
  int b;
};


int main()
{
  (void)B{.b=1}; // no warning for `BBase::x` -> NOK
}
```
https://godbolt.org/z/x5Y1EjY5E (the comments were for g++)

Compared to GCC, it also seems to ignore members of base classes, but there is still a warning about the base classes themselves. https://gcc.gnu.org/bugzilla/show_bug.cgi?id=126319 (I'm putting this here on the assumption that there is a desire to align behavior).
